Doc/library/socket.rst

```diff
@@ -66,6 +66,11 @@ created.  Socket addresses are represented as follows:
   .. versionchanged:: 3.5
      Writable :term:`bytes-like object` is now accepted.
 
+  .. versionchanged:: 3.12
+     :term:`path-like object` is now accepted.
+     Note that object which is both a :term:`path-like object` and a
+     :term:`bytes-like object` will be treated as a :term:`path-like object`.
```

Comment:
Is it important? The reversed priority may make some things simpler, for example avoid raising TypeError for bytearray or ignoring TypeError from `__fspath__()`.

Reply:
I think that we should be consistent with `os.fspath()`, one objects can't be treated as one path in `os.fspath()` and as other in `socket.bind()`. WDYT?
Also - I am not so sure if it's worth appearing in the documentation

Reply:
I'm not sure if it matters. Type that implement both buffer and path protocols and is not a subclass of `str` or `bytes` is a pretty obscure use case. If changing the priority can make the code much simpler, then we perhaps can ignore this tiny inconsistency. If the benefit is not so great, it is not worth to change this.